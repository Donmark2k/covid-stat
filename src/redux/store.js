// import { configureStore } from '@reduxjs/toolkit';
// import { createLogger } from 'redux-logger';
// import rocketReducer from './rockets/rocketSlice';
// import missionReducer from './missions/missionSlice';

// const logger = createLogger({
//   // Options for the logger can be passed here
//   // Show only the actions in the console
//   predicate: (getState, action) => action.type !== 'SOME_ACTION',
//   // Collapse the logs by default
//   collapsed: true,
// });

// export const store = configureStore({
//   reducer: {
//     missions: missionReducer,
//     rockets: rocketReducer,
//   },
//   middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
// });

// export default store;
