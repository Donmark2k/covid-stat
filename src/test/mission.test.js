import React from 'react';
import axios from 'axios';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from '../redux/store';
import { fetchMissions, joinMission } from '../redux/missions/missionSlice';
import Mission from '../components/mission';

it('Mission page renders correctly', () => {
  const tree = renderer
    .create(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Mission />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Mission redux state tests', () => {
  it('Should initially set Missions store to an empty Array', () => {
    const state = store.getState().missions;
    expect(state.missionList).toEqual([]);
  });

  it('should Join Mission payload send correct', () => {
    const expectedPayload = { payload: '9D1B7E5', type: 'missions/joinMission' };
    const actualPayload = joinMission('9D1B7E5');
    expect(actualPayload).toEqual(expectedPayload);
  });

  it('Mission fetch data from API', async () => {
    const url = 'https://api.spacexdata.com/v3/missions';
    const axiosSpy = jest.spyOn(axios, 'get');
    jest.setTimeout(90000);
    const dispatchSpy = jest.fn();

    await fetchMissions(url)(dispatchSpy);

    expect(axiosSpy).toHaveBeenCalledWith(url);
  });
});
