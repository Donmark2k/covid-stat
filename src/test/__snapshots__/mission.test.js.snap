// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission page renders correctly 1`] = `
<div
  className="px-5 pr-5"
>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
